File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/retrieverecordsfromanintersecttable3.vb
' <snippetretrieverecordsfromanintersecttable3>


 '					#Region "Retrieve records from an intersect table via LINQ"

 ' Obtain the Organization Context.
 Dim context As New OrganizationServiceContext(_serviceProxy)

 ' Create Linq Query.
 Dim roles = ( _
     From r In context.CreateQuery(Of Role)() _
     Join s In context.CreateQuery(Of SystemUserRoles)() _
     On r.RoleId Equals s.RoleId _
     Where s.SystemUserId.Equals(_userId) _
     Order By r.RoleId _
     Select r.Name)

 ' Display results.
 For Each role In roles
     Console.WriteLine("Linq Retrieved: {0}", role)
 Next role

 '					#End Region

' </snippetretrieverecordsfromanintersecttable3>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/retrieverecordsfromanintersecttable3.cs
// <snippetretrieverecordsfromanintersecttable3>


 #region Retrieve records from an intersect table via LINQ

 // Obtain the Organization Context.
 OrganizationServiceContext context = new OrganizationServiceContext(_serviceProxy);

 // Create Linq Query.
 var roles = (from r in context.CreateQuery<Role>()
              join s in context.CreateQuery<SystemUserRoles>() on r.RoleId equals s.RoleId
              where s.SystemUserId == _userId
              orderby r.RoleId
              select r.Name);

 // Display results.
 foreach (var role in roles)
 {
     Console.WriteLine("Linq Retrieved: {0}", role);
 }

 #endregion

// </snippetretrieverecordsfromanintersecttable3>